src/time.js

```diff
@@ -82,3 +108,48 @@ export function isTimeYear(i) {
   const date = i.floor(new Date(2000, 11, 31));
   return timeYear(date) >= date; // coercing equality
 }
+
+export function formatTimeTicks(scale, ticks, anchor) {
+  const format = scale.type === "time" ? timeFormat : utcFormat;
+  const template =
+    anchor === "left" || anchor === "right"
+      ? (f1, f2) => `\n${f1}\n${f2}` // extra newline to keep f1 centered
+      : anchor === "top"
+      ? (f1, f2) => `${f2}\n${f1}`
+      : (f1, f2) => `${f1}\n${f2}`;
+  switch (getTimeTicksInterval(scale, ticks)) {
+    case "millisecond":
+      return formatConditional(format(".%L"), format(":%M:%S"), template);
+    case "second":
+      return formatConditional(format(":%S"), format("%-I:%M"), template);
+    case "minute":
+      return formatConditional(format("%-I:%M"), format("%p"), template);
+    case "hour":
+      return formatConditional(format("%-I %p"), format("%b %-d"), template);
+    case "day":
+      return formatConditional(format("%-d"), format("%b"), template);
+    case "week":
+      return formatConditional(format("%-d"), format("%b"), template);
+    case "month":
+      return formatConditional(format("%b"), format("%Y"), template);
+    case "year":
+      return format("%Y");
+  }
+  throw new Error("unable to format time ticks");
+}
+
+function getTimeTicksInterval(scale, ticks) {
```

Comment:
This function being a bit difficult to grokk, I've made a chart to understand at which exact values the transitions happen.

![untitled (26)](https://github.com/observablehq/plot/assets/7001/44eda56e-c459-4dd1-bed7-06d4e0a69014)

Reply:
It’s gotta match the behavior of d3.utcTicks.